RInAppMessaging/Classes/CampaignDispatcher.swift

```diff
@@ -89,10 +99,28 @@ internal class CampaignDispatcher: CampaignDispatcherType, TaskSchedulable {
             return
         }
 
+        // fetch from imageUrl, display if successful, skip on error
+        if let resImgUrlString = campaign.data.messagePayload.resource.imageUrl, let resImgUrl = URL(string: resImgUrlString) {
+            data(from: resImgUrl) { imgBlob in
+                self.dispatchQueue.async {
```

Comment:
As this is already executed on the background IAM dispatchQueue why is it necessary to make this async?

Reply:
(Context: this line `data(from: resImgUrl) { imgBlob in` uses URLSession underneath and `URLSessionConfiguration.default`.)

I always thought the URLSession's completion returned on its own background queue so that's why I dispatched it back onto our `CampaignDispatcher.dispatchQueue` where everything else is being run off of.

Reply:
[Completion runs on a separate queue according to SO](https://stackoverflow.com/questions/48951961/how-to-tell-what-queue-a-completionhandler-executes-on). @imaginaris suggested running an experiment and this was the output:
```swift
import Foundation
func currentQueueName() -> String? {
    String(cString: __dispatch_queue_get_label(nil), encoding: .utf8)
}
let dispatchQueue = DispatchQueue(label: "IAM.CampaignDisplay", qos: .userInteractive)
dispatchQueue.async {
    print(currentQueueName()!) // output: IAM.CampaignDisplay
    URLSession.shared.dataTask(with: URL(string:"https://www.google.com")!) { x,y,z in
        print(currentQueueName()!) // output: com.apple.NSURLSession-delegate
    }.resume()
}
```
So I guess we do need to send the completion back to `CampaignDispatcher.dispatchQueue`.

Reply:
Yep, you are right